### PR TITLE
Restricting access to featured and highlighted fields 

### PR DESCRIPTION
**Pull request by katlazarova**

### CODE REVIEW

File: web/modules/custom/pineapple_general/pineapple_general.module [hunk 1]
```diff
@@ -93,6 +93,16 @@ function pineapple_general_form_alter(&$form, \Drupal\Core\Form\FormStateInterfa
           }
         }
       }
+      if (key_exists('form_id', $form_state->getBuildInfo())) {
+        $build_info = $form_state->getBuildInfo();
+        if ($build_info['form_id'] == 'node_recipe_edit_form' || $build_info['form_id'] == 'node_recipe_form') {
+          $user_roles = \Drupal::currentUser()->getRoles();
+          if (in_array('recipe_editor', $user_roles)) {
+            $form["field_r_featured"]["#access"] = FALSE;
+            $form["field_r_highlighted"]["#access"] = FALSE;
```
Comment (ElXyfer) [inline web/modules/custom/pineapple_general/pineapple_general.module:102]: Use single quotes.

File: web/modules/custom/pineapple_general/pineapple_general.module [hunk 2]
```diff
@@ -93,6 +93,16 @@ function pineapple_general_form_alter(&$form, \Drupal\Core\Form\FormStateInterfa
           }
         }
       }
+      if (key_exists('form_id', $form_state->getBuildInfo())) {
+        $build_info = $form_state->getBuildInfo();
+        if ($build_info['form_id'] == 'node_recipe_edit_form' || $build_info['form_id'] == 'node_recipe_form') {
```
Comment (ElXyfer) [inline web/modules/custom/pineapple_general/pineapple_general.module:98]: Why are you checking for the form Id again when you're already in a switch statement that's already checked for those specific form ids?

File: web/modules/custom/pineapple_general/pineapple_general.module (hunk 1 above)
Comment (katlazarova) [reply web/modules/custom/pineapple_general/pineapple_general.module]: Done

File: web/modules/custom/pineapple_general/pineapple_general.module (hunk 2 above)
Comment (katlazarova) [reply web/modules/custom/pineapple_general/pineapple_general.module]: Fair point, it was an oversight. Done.

File: web/modules/custom/pineapple_general/pineapple_general.module [hunk 3]
```diff
@@ -93,6 +93,13 @@ function pineapple_general_form_alter(&$form, \Drupal\Core\Form\FormStateInterfa
           }
         }
       }
+      if (key_exists('form_id', $form_state->getBuildInfo())) {
```
Comment (ElXyfer) [inline web/modules/custom/pineapple_general/pineapple_general.module:96]: Again, why are you checking if the form Id key exists when your entire switch statement is based on the form id and you're inside it?

Comment (katlazarova) [reply web/modules/custom/pineapple_general/pineapple_general.module]: I'm sorry I didn't notice. The entire function was somewhere else and when I moved it, I accidentally left those if statements there.

Comment (katlazarova) [reply web/modules/custom/pineapple_general/pineapple_general.module]: It's done

Comment (ElXyfer) [reply web/modules/custom/pineapple_general/pineapple_general.module]: Read what the code you’re copying does and also the code in surrounding areas these kinds of mistakes shouldn’t be happening if you’re paying attention to what you’re doing.

**Pull request merged**